### PR TITLE
Add password recovery mechanism

### PR DESCRIPTION
**Pull request by matthewdunsdon**
- Introduced a password reset code on the backend
- Refactored email code so that emails can be "sent" either from dev boxes or Azure servers
- Introduced code for email confirmation (feature turned off)
- Moved account services to use angular's promises
- Fixed jshint issues in files that were modified

### CODE REVIEW

**Comment by JDawes-ScottLogic** [comment]
thumbs-up-bruce-lee_2

File: VotingApplication/VotingApplication.Web/Api/Controllers/AccountController.cs [hunk 1]
```diff
@@ -153,6 +174,64 @@ public IHttpActionResult Logout()
...
+            {
+                ModelState.AddModelError("", "The user either does not exist or is not confirmed.");
+                return BadRequest(ModelState);
+            }
+
+            string code = await UserManager.GeneratePasswordResetTokenAsync(user.Id);
+
+            var callbackUrl = Url.Content("~/#/Account/ResetPassword?email=") + HttpUtility.UrlEncode(user.Email) + "&code=" + HttpUtility.UrlEncode(code);
+
+            await UserManager.SendEmailAsync(user.Id, "Pollster Password Reset", "Please reset your password by clicking <a href=\"" + callbackUrl + "\">here</a>");
```
Comment (tpkelly) [inline VotingApplication/VotingApplication.Web/Api/Controllers/AccountController.cs]: General rule of thumb, but you should avoid "click here" links: http://uxmovement.com/content/why-your-links-should-never-say-click-here/

I'll push this through and fix in a separate commit though.

File: VotingApplication/VotingApplication.Web/Api/Controllers/AccountController.cs [hunk 2]
```diff
@@ -328,15 +407,30 @@ public IEnumerable<ExternalLoginViewModel> GetExternalLogins(string returnUrl, b
...
             }
 
+            // Commented out code for sending confirmation emails
+            /*
+            string code = await UserManager.GenerateEmailConfirmationTokenAsync(user.Id);
+            var callbackUrl = Url.Link("ConfirmEmail", new { userId = user.Id, code = code });
+
+            string body = String.Format("Your account has been successfully created, but you still need one more step.<br />Click <a href={0}> here </a>to verify your account", callbackUrl);
+            await UserManager.SendEmailAsync(user.Id, "Pollster account confirmation", body);
+            */
```
Comment (tpkelly) [inline VotingApplication/VotingApplication.Web/Api/Controllers/AccountController.cs]: Do we still need this?

**Comment by tpkelly** [comment]
Recreating from my branch

**Pull request closed without merging**